src/main/java/dinkplugin/notifiers/GroupStorageNotifier.java

```diff
@@ -0,0 +1,192 @@
+        if (event.getGroupId() != GROUP_STORAGE_LOADER_ID)
+            return;
+
+        Widget widget = client.getWidget(GROUP_STORAGE_LOADER_ID, 1);
+        if (widget == null)
+            return;
+
+        if (isEnabled() && StringUtils.containsIgnoreCase(widget.getText(), "Saving")) {
+            ItemContainer inv = getInventory();
+            if (inv != null) {
+                Map<Integer, Integer> modified = reduce(inv.getItems());
+                Map<Integer, Integer> delta = computeDifference(initial, modified);
+                if (!delta.isEmpty()) {
+                    handleNotify(delta);
+                }
+            }
+        }
+
+        initial = Collections.emptyMap();
+    }
+
+    private void handleNotify(Map<Integer, Integer> inventoryChanges) {
+        // Calculate transaction information
+        List<SerializedItemStack> deposits = new ArrayList<>();
+        List<SerializedItemStack> withdrawals = new ArrayList<>();
+        long debits = 0, credits = 0;
+        for (Map.Entry<Integer, Integer> entry : inventoryChanges.entrySet()) {
+            int diff = entry.getValue(); // positive=withdraw, negative=deposit
+            SerializedItemStack item = ItemUtils.stackFromItem(itemManager, entry.getKey(), Math.abs(diff));
+            long stackPrice = item.getTotalPrice();
+            if (diff < 0) {
+                deposits.add(item);
+                debits += stackPrice;
+            } else {
+                withdrawals.add(item);
+                credits += stackPrice;
+            }
+        }
+        long netValue = debits - credits;
+
+        // Ensure transaction is large enough to be logged
+        if (debits < config.bankMinValue() && credits < config.bankMinValue())
+            return;
+
+        // Sort lists so more valuable item transactions are at the top
+        Comparator<SerializedItemStack> valuable = Comparator.comparingLong(SerializedItemStack::getTotalPrice).reversed();
+        deposits.sort(valuable);
+        withdrawals.sort(valuable);
+
+        // Convert lists to strings
+        BiFunction<Collection<SerializedItemStack>, String, String> formatItems = (items, linePrefix) -> {
+            if (items.isEmpty()) return EMPTY_TRANSACTION;
+            return items.stream()
+                .map(ItemUtils::formatStack)
+                .collect(Collectors.joining('\n' + linePrefix, linePrefix, ""));
+        };
+        String depositString = formatItems.apply(deposits, "+ ");
+        String withdrawalString = formatItems.apply(withdrawals, "- ");
+
+        // Build content
+        String playerName = client.getLocalPlayer().getName();
+        String content = StringUtils.replaceEach(config.bankNotifyMessage(),
+            new String[] { "%USERNAME%", "%DEBITS%", "%CREDITS%" },
+            new String[] { playerName, depositString, withdrawalString }
+        );
+        String formattedText = config.discordRichEmbeds() ? Field.formatBlock("diff", content) : content;
```

Comment:
Using a code block in the embed body (rather than fields) is new. Is it too inconsistent from the other messages that the full body is wrapped in a code block? Should we have two code blocks instead (just surrounding the item lists)?

Reply:
I think this is fine :+1: